### DISCUSSION TITLE
Conditional UI based on Query/Database

### DISCUSSION THREAD

**Discussion by Mozix** [Budibase/budibase#12327] [Help] [answered]
Hello,

Would it be possible to do the following with Budibase:

First I have a table that tells me which event should have what options

Now I wanna make it so the query returns me all the options for an event

Then I wanna create components like text fields or whatever based on the returned data

So if I chose event1

I should be able to input:
Option1 and option2

If I chose event2
I should be able to input:
Option1 and option3

And so on...

If there's a solution to this I'd be happy!

Thank you

**Comment by melohagan** [comment, marked as answer]
Hey @Mozix 

> Then I wanna create components like text fields or whatever based on the returned data

This is currently not easily possible, but has been requested: https://github.com/Budibase/budibase/issues/10860

You could potentially do this through conditional UI: https://docs.budibase.com/docs/conditions
but it could be quite tricky and not very scalable.
